### PR TITLE
Fix: normalizer ensuring type is always present in TransactionParams

### PR DESCRIPTION
**Pull request by vinistevam**
## Explanation

This PR aims to ensure `type` is always present in the `TransactionParams` by matching the [old Extension Normalizer](https://github.com/MetaMask/metamask-extension/blob/7dee5ba77e658a5856d242f4f4f0b998bcf8ae99/app/scripts/controllers/transactions/lib/util.js#L23).

### `@metamask/transaction-controller`

- **FIXED**: the `type` normalizer to use the `addHexPrefix` and do not set the `type` as `undefined`.

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Comment by vinistevam** [comment]
@metamaskbot publish-preview

File: packages/transaction-controller/src/TransactionController.test.ts [hunk 1]
```diff
@@ -2006,7 +2006,6 @@ describe('TransactionController', () => {
         gasPrice: '0x1',
         to: ACCOUNT_MOCK,
         value: '0x0',
-        type: TransactionType.simpleSend,
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.test.ts:2009]: Good spot 😄  It's a fun one having a `type` property in both the parameters and the options.

File: packages/transaction-controller/src/utils/utils.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ const NORMALIZERS: { [param in keyof TransactionParams]: any } = {
     addHexPrefix(maxPriorityFeePerGas),
   estimatedBaseFee: (maxPriorityFeePerGas: string) =>
     addHexPrefix(maxPriorityFeePerGas),
-  type: (type: string) => (type === '0x0' ? '0x0' : undefined),
+  type: (type: string) => addHexPrefix(type),
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/utils/utils.ts:34]: In a future PR, we could go one step further and always derive a `type` even if one wasn't provided by looking at whether there is `gasPrice` or `maxFeePerGas` etc.

That's what we already do on submission using `isEIP1559Transaction`.

Perhaps worth confirming by testing an older commit with the extension controller, but my understanding is if a dapp transaction doesn't specify type, it won't be in the parameters either.

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

File: packages/transaction-controller/src/utils/utils.ts (hunk 2 above)
Comment (FrederikBolding) [reply packages/transaction-controller/src/utils/utils.ts:34]: I think this would be a good addition yeah! 👍

**Review by FrederikBolding — APPROVED** [review]
Can confirm this fixed in the snaps issue!

**Pull request merged**